### PR TITLE
Set up JS linting for repo

### LINKED ISSUE #102: Set up JS linting for repo
Set up JS linting via ESLint for this repository to highlight formatting issues as per WP coding standards.

### PR DESCRIPTION
**Pull request by spacedmonkey**
### Description of the Change

As documented in the WordPress handbook, wp-scripts, as lots of handy linters and checkers. [See](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#setup). Copy and pasted the format npm command in our project and run it. This is the result. Going forward developers can use this command to lint js/md/yml files. 

Closes #102

### Changelog Entry

> Added - New feature
> Changed - Existing functionality
> Deprecated - Soon-to-be removed feature
> Removed - Feature
> Fixed - Bug fix
> Security - Vulnerability

### Credits

Props @spacedmonkey

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+	root: true,
+	extends: [ 'plugin:@wordpress/eslint-plugin/recommended' ],
+	plugins: [ 'import' ],
+	globals: {
+		wp: 'off',
+		ajaxurl: 'readonly',
+		FormData: 'readonly',
```
Comment (spacedmonkey) [inline .eslintrc.js:8]: When we revisit this, we need to import this correctly.

File: package.json [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
+	"devDependencies": {
+		"@wordpress/env": "^5.7.0",
+		"@wordpress/scripts": "^24.6.0"
+	},
+	"scripts": {
+		"wp-env": "wp-env",
+		"format": "wp-scripts format",
+		"lint:js": "wp-scripts lint-js",
+		"lint:md:docs": "wp-scripts lint-md-docs",
+		"packages-update": "wp-scripts packages-update",
```
Comment (felixarntz) [inline package.json]: What do these do? Do we really need them? Seems a bit beyond the scope of this issue.

File: .github/workflows/php-lint.yml [hunk 3]
```diff
@@ -1,58 +1,58 @@
...
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-composer-
+            - name: Validate Composer configuration
+              run: composer validate --strict
+            - name: Composer Install
+              run: composer install --no-interaction --no-progress
+            - name: PHP Lint
+              run: composer lint
```
Comment (felixarntz) [inline .github/workflows/php-lint.yml]: Why all the indentation changes to the `.yml` files? That seems unrelated to JS linting?

File: .github/workflows/php-test.yml [hunk 4]
```diff
@@ -1,59 +1,59 @@
...
+                  node-version-file: '.nvmrc'
+                  cache: npm
+            - name: npm install
+              run: npm ci
+            - name: Install WordPress
+              run: npm run wp-env start
+            - name: Running php unit tests
+              run: npm run test-php
+            - name: Running multisite unit tests
+              run: npm run test-php-multisite
```
Comment (felixarntz) [inline .github/workflows/php-test.yml]: Same question here.

**Review by felixarntz — COMMENTED** [review]
Thanks @spacedmonkey! Mostly looks great, just a few small questions.

File: package.json (hunk 2 above)
Comment (spacedmonkey) [reply package.json]: I can remove it.

File: .github/workflows/php-lint.yml (hunk 3 above)
Comment (spacedmonkey) [reply .github/workflows/php-lint.yml]: It is part of the `npm run format`. I dont see a problem with this change.

Comment (felixarntz) [reply .github/workflows/php-lint.yml]: It's not a JS formatting change though, so I'd argue it goes beyond the scope of the issue.

But more importantly, this change conflicts with what is defined in our `.editorconfig` file. Whatever of the two formatting options we go with, we need to consistently handle it.

**Comment by spacedmonkey** [comment]
@felixarntz Feedback complete.

File: package.json [hunk 5]
```diff
@@ -11,6 +11,8 @@
   },
   "scripts": {
     "wp-env": "wp-env",
+    "lint:js": "wp-scripts lint-js",
+    "lint:js:fix": "npm run lint:js -- --fix",
```
Comment (felixarntz) [inline package.json]: Can we use syntax consistent with the other existing scripts (`-` instead of `:`)?

Also, the convention for formatting scripts is to use `format`, so I think this would be better named as `format-js`.
```suggestion
    "lint-js": "wp-scripts lint-js",
    "format-js": "npm run lint:js -- --fix",
```

**Review by felixarntz — COMMENTED** [review]
@spacedmonkey Two naming suggestions here, but code-wise looks good.

File: package.json (hunk 5 above)
Comment (spacedmonkey) [reply package.json]: Fixed.

**Review by jjgrainger — APPROVED** [review]
Thanks @spacedmonkey approved! This will be great to get this merged in today once @felixarntz has approved as this will have a positive impact on the remaining work 👍

**Review by felixarntz — APPROVED** [review]
Thanks @spacedmonkey!
